### ISSUE TITLE
importKind not present in ImportDeclaration TypeScript Definition

### ISSUE THREAD

**Issue by kube** [babel/babel#9485]
This Flow syntax:

```ts
import type { SomeType } from 'moduleName';
```

gives an `ImportDeclaration` with `importKind` to `"type"` at runtime.

**@babel/types** TypeScript definition does not provide `importKind` property definition:

```ts
export interface ImportDeclaration extends BaseNode {
  type: "ImportDeclaration";
  specifiers: Array<ImportSpecifier | ImportDefaultSpecifier | ImportNamespaceSpecifier>;
  source: StringLiteral;
}
```

**Comment by danez** [comment]
That is because typescript does not have a different syntax for type imports and regular imports afaik. Which in turn means our parser and compiler does not know if a import is a type or a valid js structure.

What our typescript transform is doing is simply removing all type annotations and typescript constructs and then checking which imports are unused and removes them.

**Comment by kube** [comment]
@danez Sorry for not being clear about the issue.

What I'm doing is a Transpiler from FlowType to TypeScript.

Indeed there's no `importKind` in TypeScript, but there is in FlowType.

So I need to transpile this FlowType:

```ts
import type { Animal } from './Animal';
```

to this TypeScript:

```ts
import { Animal } from './Animal';
```

The Visitor is quite simple here, but I need to cast my node to `any` to be able to access `importKind`:

```ts
  ImportDeclaration(path) {
    // Need to cast here, as importKind does not exist as type property of node
    // But exists at runtime
    if ((path.node as any).importKind === "type") {
      path.replaceWith(
        t.importDeclaration(
          // ...
        )
      );
    }
  }
```

****

**To sum up**: `importKind` should be added in `ImportDeclaration`

**Comment by danez** [comment]
Okay yes there was  a misunderstanding. I understand now. Will look into it.

**Comment by danez** [comment]
I created a PR at DefinitelyTyped/DefinitelyTyped#32948, you can track the progress there.

**Comment by wincent** [comment]
@danez: What ended up happening here? I see you closed the DefinitelyTyped PR without comment.

**Comment by danez** [comment]
This is fixed as of version 7.3.3. The PR for DefinitelyTyped was not necessary as we autogenerate and include the types in the repo now. It was fixed by #9490.
